src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -24,21 +25,43 @@ internal abstract class AbstractTodoCommentService : ITodoCommentService
         protected abstract int GetCommentStartingIndex(string message);
         protected abstract void AppendTodoComments(ImmutableArray<TodoCommentDescriptor> commentDescriptors, SyntacticDocument document, SyntaxTrivia trivia, ArrayBuilder<TodoComment> todoList);
 
-        public async Task<ImmutableArray<TodoComment>> GetTodoCommentsAsync(
+        public async Task<ImmutableArray<TodoCommentData>> GetTodoCommentDataAsync(
+            Document document,
+            ImmutableArray<TodoCommentDescriptor> commentDescriptors,
+            CancellationToken cancellationToken)
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
```

Comment:
always attempt to go to OOP.  We're potentially parsing all of the files in teh solution.  we don't want to do that in-proc.